src/EnergyPlus/ConvectionCoefficients.cc

```diff
@@ -1321,6 +1323,7 @@ void GetUserConvectionCoefficients(EnergyPlusData &state)
                     if (HcExt_ConvectionTypesMap.find(equationName) != HcExt_ConvectionTypesMap.end()) {
                         ExtValue = HcExt_ConvectionTypesMap.at(equationName);
 
+                        // TODO: Use BITF Comparison
```

Comment:
Noting this here now FFR -- I'll try changing this one again later once I've fully converted these to an ```enum class```. I was getting a ```Wshift-count-overflow``` warning here, and it was causing example files to crash.

Warning
```
/Users/mmitchel/Projects/EnergyPlus/dev/EnergyPlus/src/EnergyPlus/ConvectionCoefficients.cc:1327:43: warning: shift count >= width of type [-Wshift-count-overflow]
                                          BITF(ConvectionConstants::HcExt_ASHRAESimpleCombined) | BITF(ConvectionConstants::HcExt_TarpHcOutside) |
                                          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/mmitchel/Projects/EnergyPlus/dev/EnergyPlus/src/EnergyPlus/BITF.hh:55:20: note: expanded from macro 'BITF'
#define BITF(B) (1 << (int(B)))
                   ^
/Users/mmitchel/Projects/EnergyPlus/dev/EnergyPlus/src/EnergyPlus/BITF.hh:56:38: note: expanded from macro 'BITF_TEST_ANY'
#define BITF_TEST_ANY(V, B) (((V) & (B)) != 0)
                                     ^
```

Proposed code change
```
if (BITF_TEST_ANY(BITF(ExtValue),
                                 BITF(ConvectionConstants::HcExt_ASHRAESimpleCombined) | 
                                 BITF(ConvectionConstants::HcExt_TarpHcOutside) |
                                 BITF(ConvectionConstants::HcExt_MoWiTTHcOutside) | 
                                 BITF(ConvectionConstants::HcExt_DOE2HcOutside) |
                                 BITF(ConvectionConstants::HcExt_AdaptiveConvectionAlgorithm)))
```

Reply:
So ... the `BITF` trick only works for enumerated values less than 64. I see that although there are fewer than 64 enumerations right now, they are not numbered compactly. Once again, is this necessary? Doing integer logic on enumerations is not good programming practice.

Reply:
Ah, I did not know that. I'm going to revert that commit and apply these once the enum conversion happens. 

And no, it's not necessary. I'll be getting rid of all the integer logic as this moves forward.

Reply:
Yeah. The `BITF` trick uses the bits in a 64-bit integer to represent the enumerations. Bit 0 represents the 0th enumerated value, bit 1 represents the 1st, etc. So you construct at compile time an integer that has all of the bits corresponding to the enumerations you are interested in comparing to set and then see if the current value hits one of those bits. It's basically a fast implementation `std::vector<bool>(64)`. FWIW `switch/case` does not have this limitation.